### PR TITLE
new option to store session in SQL database

### PR DESCRIPTION
**Pull request by briskt**
### Added
- Conditional creation of RDS or Elasticache based on the value of `session_store_type` variable.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -124,6 +130,43 @@ resource "aws_elasticache_cluster" "memcache" {
...
+resource "random_password" "db_root" {
+  count = var.session_store_type == "sql" ? 1 : 0
+
+  length = 16
+}
+
+/*
+ * Create RDS database for session store, if session_store_type is "sql"
+ */
+module "rds" {
```
Comment (Baggerone) [inline main.tf:145]: I'm a little confused.
If ssp-base has its own database, why do we need to create and RDS database for the IdP Hub?

Comment (briskt) [reply main.tf:145]: ssp-base is set up to *use* a database, but since the hub runs on its own, apart from idp-in-a-box, it needs its own database.

Comment (Baggerone) [reply main.tf:145]: I finally found it. It looks like silauth is what wants to use the ssp-base database.  I see there are migrations in silauth.  I haven't tracked down yet where/when they are run.

**Review by Baggerone — APPROVED** [review]
I guess we still have a db migration to plan out or at least some way of setting up the database table.

**Comment by Baggerone** [comment]
> > I guess we still have a db migration to plan out or at least some way of setting up the database table.
> 
> simpleSAMLphp creates the table

Oh, nice.

**Pull request merged**